server/schedule/sched_limiter.go

```diff
@@ -0,0 +1,333 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schedule
+
+import (
+	"container/list"
+	"sync"
+	"time"
+
+	"github.com/pingcap/kvproto/pkg/eraftpb"
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/kvproto/pkg/pdpb"
+	"github.com/pingcap/pd/server/core"
+	log "github.com/sirupsen/logrus"
+)
+
+var historyKeepTime = 5 * time.Minute
+
+// HeartbeatStreams is an interface of async region heartbeat.
+type HeartbeatStreams interface {
+	SendMsg(region *core.RegionInfo, msg *pdpb.RegionHeartbeatResponse)
+}
+
+// SchedLimiter is used to limit the speed of scheduling.
+type SchedLimiter struct {
```

Comment:
Seems not a good name. Looks it is not only used for limit schedule speed, but also manages all pending operators, including dispatch and drop. How about rename to something like `OperatorManager` or `OperatorController`?